smartapps/tonesto7/nest-manager.src/nest-manager.groovy

```diff
@@ -777,36 +778,38 @@ private sendNestApiCmd(cmdTypeId, cmdType, cmdObj, cmdObjVal, childId) {
 
 void workQueue() {
 	//log.trace "workQueue..."
+    def cmdDelay = !atomicState.cmdDelayVal ? 4 : atomicState?.cmdDelayVal.toInteger()
     def cmdQueue = !atomicState?.cmdQ ? [] : atomicState?.cmdQ
     try {
-    	def cmdDelay = !atomicState.cmdDelayVal ? 4 : atomicState?.cmdDelayVal.toInteger()
-		if(cmdQueue?.size() > 0) { 
-        	cmdProcState(false)
             if(cmdQueue.size() > 0) {
-            	def cmd = cmdQueue?.first()
+                cmdQueue = atomicState?.cmdQ
+            	def cmd = cmdQueue?.remove(0)
```

Comment:
This is interesting to me... How would that work?  You are declaring a remove as an array